hypothesis/via
# Consider the resource type and referrer to get the URL to check

This attempts to distinguish between a page, resources in that page and other items to check what to run through checkmate.

We list the root pages, but not all sub-domains and CDNs where other content might be coming from. This means we need to be more lenient with sub-resources, but we should still check they come from somewhere trusted to the extent we can.

## The logic

The basic idea is to:

* Consider both the URL and the referrer
* Decide which came from Via, raw pages or easily identifiable sub-resources
* We should check pages properly against the allow list
* Sub-resources are ok to check only against the block list, assuming we check the parent page against the allow list

## Testing notes

 * To really test this you need checkmate actually enforcing the allow list
 * Checkout `allow-list` in checkmate and run `make services dev`
 * Run `make dev` here
 * Add some print statements in `blocker.py`?
 * Check some pages like: http://localhost:9080/https://www.theguardian.com/uk
